### PR TITLE
Fix NULL values in date columns

### PR DESCRIPTION
**Pull request by davidmanzanares**
Fix for:
- [x] https://github.com/CartoDB/carto-vl/issues/508 solved by https://github.com/CartoDB/carto-vl/pull/509/files#diff-eef002f2ff8de5af0ca03149dcd70afbL503
- [x] https://github.com/CartoDB/carto-vl/issues/517 solved by https://github.com/CartoDB/carto-vl/pull/509/files#diff-5bc975a5d8a5eecf453406f291624acbR57
- [x] https://github.com/CartoDB/carto-vl/issues/409 solved by checking for NaNs in the GLSL code

Easily testable with http://127.0.0.1:8080/examples/editor/#eyJhIjoibW5tYXBwbHV0b19udWxsZGF0ZXMiLCJiIjoiIiwiYyI6ImNhcnRvdmwiLCJkIjoiaHR0cHM6Ly97dXNlcn0uY2FydG8uY29tIiwiZSI6ImNvbG9yOiByYW1wKGxpbmVhcihsb2coJG51bWZsb29ycyksIDEsIDQpLCBFYXJ0aClcbnN0cm9rZUNvbG9yOiBvcGFjaXR5KHdoaXRlLCAwLjIpXG5maWx0ZXI6IHRvcnF1ZSgkYnVpbHRfdGltZSwgMTAsICBmYWRlKDAsMTAwMDApKSAiLCJmIjp7ImxuZyI6LTczLjk4NzY3MjMxNTg2OTgyLCJsYXQiOjQwLjc0MDgyOTc5MTkzMzE4fSwiZyI6MTEuODgzMzc3NzE2MTM3MTMzLCJoIjoiRGFya01hdHRlciJ9

### CODE REVIEW

**Comment by davidmanzanares** [comment]
hahahaha, yes, we should fix this with https://github.com/CartoDB/carto-vl/issues/409

**Comment by davidmanzanares** [comment]
I decided to include in this branch the fix for https://github.com/CartoDB/carto-vl/issues/517

File: src/core/viz/expressions/torque.js [hunk 1]
```diff
@@ -143,11 +143,15 @@ export class Torque extends BaseExpression {
     }
     eval(feature) {
         const input = this.input.eval(feature);
+        if (Number.isNaN(input)){
```
Comment (IagoLast) [inline src/core/viz/expressions/torque.js]: we should document the behaviour for null/nan values

**Comment by IagoLast** [comment]
LGTM, just a minor doc issue

**Comment by davidmanzanares** [comment]
@jgoizueta , pls, re-review

File: src/core/viz/expressions/torque.js (hunk 1 above)
Comment (davidmanzanares) [reply src/core/viz/expressions/torque.js]: yep

File: src/core/viz/expressions/torque.js [hunk 2]
```diff
@@ -193,7 +197,20 @@ export class Torque extends BaseExpression {
...
+        this.preface = `
+        #ifndef TORQUE
+        #define TORQUE
+        float torque(float _input, float cycle, float duration, float fadeIn, float fadeOut){
+            float x = 0.;
+            // Check for NaN
+            if (_input <= 0.0 || 0.0 <= _input){
+                x = 1.- clamp(abs(_input-cycle)*duration/(_input>cycle? fadeIn: fadeOut), 0.,1.);
+            }
+            return x;
```
Comment (jgoizueta) [inline src/core/viz/expressions/torque.js]: So... now NaN in the input will produce a NaN result in torque, which will not match any value... basically ignoring the input, right?

Comment (davidmanzanares) [reply src/core/viz/expressions/torque.js]: Hmu... No, a NaN in the input will result in a `0` in torque. See that `x` is initialized to `0`

Comment (jgoizueta) [reply src/core/viz/expressions/torque.js]: Oh, I see. so 0 being a non-match it makes sense

**Pull request merged**